docs/working-with-streams.md

```diff
@@ -1,27 +1,47 @@
+To support a variety of use cases efficiently, Timeplus offers multiple types of streams:
 
-## Create a stream
-You can create a stream via the Timeplus Console UI, or via [SQL](/sql-create-stream). When you [ingest data](/ingestion) into Timeplus from Kafka or file sources, streams can be created automatically to match the schema of the data.
+1. **Append Stream**
+   The default stream type in Timeplus. It uses columnar encoding and is optimized for **range scans** via a **sorting key**. It suits workloads with infrequent data mutations (e.g., `UPDATE` or `DELETE`).
 
-## Query a stream
+2. **Mutable Stream**
+   Row-encoded and similar in behavior to a **MySQL table**, where each primary key corresponds to a single row. It is optimized for **frequent data mutations** (`UPDATE`, `UPSERT`, `DELETE`) and supports **point and range queries** via **primary or secondary indexes**.
 
-By default, querying the stream will continuously scan new events and output new results. It never ends unless the user cancels the query. For example, you can get the latest web logs with HTTP 500 error or get the min/max/avg of a metric for every minute from an IoT device. Please read [Streaming Queries](/stream-query) for more details.
+3. **Versioned Key-Value Stream**
+   Similar to the mutable stream but uses **columnar encoding**. It offers better **compression** but lower performance for updates and point queries, especially when cardinality is high. Best suited for scenarios where **data mutations are less frequent**.
 
-If you only want to analyze the existing data and need an immediate response, you can run [Non-streaming Queries](/history) via the [table](/functions_for_streaming#table) function. This will turn the query in the bounded mode and only scan the existing data. For example, you can run `select count(*) from table(stream1)` to get the total number of rows in the data stream.
+4. **Changelog Stream**
+   Designed to model **change data capture (CDC) events**, with **columnar encoding** for efficient downstream processing.
 
+5. **External Stream**
+   As the name implies, the data resides outside of Timeplus. Timeplus can reference external sources (e.g., a **Kafka topic**) and execute **streaming SQL** queries against them in real time.
 
+> Note: Timeplus also supports [External Tables](/sql-create-external-table), which allow **historical queries and inserts** only (e.g., against ClickHouse, MySQL, PostgreSQL, MongoDB, etc.).
 
-## Delete a stream
 
-From the web console, you can delete the stream. This will permanently delete all data in the stream and delete the stream itself. Data cannot be recovered after deletion.
+## Stream Internals
+
+When users [create a stream](/sql-create-stream) in Timeplus, they can specify the number of **shards** for the stream. Each shard consists of two core components at the storage layer:
+
+1. **Streaming Store**
```

Comment:
Store or Storage? Let's be consistent. Last time, I checked with Wei, we use XYZ storage for website/graphics

Reply:
also the arch is Store and usually in tech, we use store